engine/engine/src/engine.cpp

```diff
@@ -275,6 +275,8 @@ namespace dmEngine
             dmResource::Release(engine->m_Factory, engine->m_MainCollection);
         dmGameObject::PostUpdate(engine->m_Register);
 
+        dmGameObject::DeleteRegister(engine->m_Register); // Delete all collections and game objects
```

Comment:
Not sure why this wasn't noticed before.
I think it seems like a good idea to destroy the collections/game objects before moving on to destroy the other engine systems.